### PR TITLE
PMP: Fix and enhance overlap detection in `merge_duplicate_vertices_on_boundary_cycles`

### PR DESCRIPTION
**Pull request by MaelRL**
## Summary of Changes

Bug fix: improper check of overlapping intervals.
Enhancement: don't throw everything as soon as there is a pair of intervals that are incompatible, only remove one of the intervals (taken arbitrarily as the widest).

## Release Management

* Affected package(s): `Polygon_mesh_processing`

### CODE REVIEW

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/merge_border_vertices.h [hunk 1]
```diff
@@ -58,6 +61,82 @@ struct Less_on_point_of_target
   const VertexPointMap& vpm;
 };
 
+template <typename VPM, typename PolygonMesh>
```
Comment (sloriot) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/merge_border_vertices.h]: Add a comment explaining the main idea?

Comment (MaelRL) [reply Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/merge_border_vertices.h]: There's a comment where the function is called... but I see that the comment is now outdated. I'll update it.

**Comment by maxGimeno** [comment]
I think there are more conversion warnings https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.1-Ic-104/Polyhedron_Demo/TestReport_afabri_x64_Cygwin-Windows10_MSVC2015-Debug-64bits.gz

**Comment by MaelRL** [comment]
See a91a225 above

**Comment by maxGimeno** [comment]
I looked but I thought it was others.

**Comment by MaelRL** [comment]
You're right, sorry: the previous fix was wrong.

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.1-Ic-108.shtml

**Comment by MaelRL** [comment]
@sloriot you messed up that merge, this targets 5.0

**Comment by sloriot** [comment]
Fixed

**Pull request merged**